### PR TITLE
API-2945 Try to make it publish

### PR DESCRIPTION
**Pull request by Tobias1087**
API-2945 Updates to register onto the API Platform

### CODE REVIEW

File: app/uk/gov/hmrc/apinotificationpull/connectors/ServiceLocatorConnector.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  val handlerOK: () => Unit = () => Logger.info("Service is registered on the service locator")
+  val handlerError: Throwable => Unit = e => Logger.error("Service could not register on the service locator", e)
+  val metadata: Option[Map[String, String]] = Some(Map("third-party-api" -> "true"))
+
+  def register(): Future[Boolean] = {
+    implicit val hc: HeaderCarrier = new HeaderCarrier
+
+    val registration = Registration(appName, appUrl, metadata)
+    http.POST(s"$serviceUrl/registration", registration, Seq("Content-Type" -> "application/json")) map {
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationpull/connectors/ServiceLocatorConnector.scala]: You could reuse the Play constants:
- CONTENT_TYPE
- JSON

See here:
https://github.com/hmrc/api-notification-pull/blob/e77a480e292c74c78adde7fb798591afd1e2907d/app/uk/gov/hmrc/apinotificationpull/controllers/DefinitionController.scala#L32

File: build.sbt [hunk 2]
```diff
@@ -32,7 +32,7 @@ lazy val appDependencies: Seq[ModuleID] = compile ++ test
 
 lazy val compile = Seq(
   ws,
-  "uk.gov.hmrc" %% "bootstrap-play-25" % "1.2.0"
+  "uk.gov.hmrc" %% "bootstrap-play-25" % "1.2.0-0-g0000000"
 )
```
Comment (googley42) [inline build.sbt]: version `bootstrap-play-25" % "1.2.0-0-g0000000` looks a bit odd

Comment (gokyo) [reply build.sbt]: I assume Toby is still testing now.
Perhaps he will change it to `1.3.0` after `bootstrap-play-25` is released :)

File: test/unit/uk/gov/hmrc/apinotificationpull/controllers/ServiceLocatorConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+      val registration =
+        Registration(serviceName = "api-notification-pull",
+          serviceUrl = "http://api-notification-pull.service",
+          metadata = Some(Map("third-party-api" -> "true")))
+
+      when(mockHttpClient.POST(
+        eqs(s"$serviceUrl/registration"),
+        eqs(registration),
+        eqs(Seq("Content-Type" -> "application/json"))
```
Comment (gokyo) [inline test/unit/uk/gov/hmrc/apinotificationpull/controllers/ServiceLocatorConnectorSpec.scala]: You could use the Play constants here for the content type header name and value

**Pull request merged**